### PR TITLE
Orv2 1273 sftp v1

### PR DESCRIPTION
**Pull request by gchauhan-aot**
# Description

This PR is to send the CGI file to sftp server. 

Note: the controller is there for now to facilitate testing and I will remove it once the create file functionality is ready and scheduled.
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A : Tested manually
- [ ] Test B

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1384-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1384-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1384-dops.apps.silver.devops.gov.bc.ca/api)
- [Scheduler](https://onroutebc-1384-scheduler.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1384-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1384-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1384-dops.apps.silver.devops.gov.bc.ca/api)
- [Scheduler](https://onroutebc-1384-scheduler.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

### CODE REVIEW

File: scheduler/src/modules/cgi-sftp/cgi-sftp.service.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+@Injectable()
+export class CgiSftpService {
+  upload(fileData: Express.Multer.File, fileName: string) {
+    const sftp = new Client();
+    const connectionInfo: Client.ConnectOptions = getSFTPConnectionInfo();
+    const remotePath = process.env.CFS_REMOTE_PATH;
+    sftp
+      .connect(connectionInfo)
+      .then(() => {
+        console.log('writing file', remotePath + fileName);
```
Comment (krishnan-aot) [inline scheduler/src/modules/cgi-sftp/cgi-sftp.service.ts]: Can console.log be swapped with logger?

File: docker-compose.yml [hunk 2]
```diff
@@ -1,4 +1,13 @@
 services:
+  sftp:
+    container_name: sftp
+    image: atmoz/sftp
+    volumes:
+        - C:\Users\chauh\.ssh\share:/home/f3535t/data
+        - C:\Repository\keys\cfs_key.pub:/home/f3535t/.ssh/keys/cfs_key.pub:ro
```
Comment (krishnan-aot) [inline docker-compose.yml]: Is this correct? It is pointing to your local?

File: docker-compose.yml [hunk 3]
```diff
@@ -191,6 +200,12 @@ services:
       TPS_PENDING_POLLING_INTERVAL: ${TPS_PENDING_POLLING_INTERVAL}
       TPS_ERROR_POLLING_INTERVAL: ${TPS_ERROR_POLLING_INTERVAL}
       TPS_MONITORING_POLLING_INTERVAL: ${TPS_MONITORING_POLLING_INTERVAL}
+      CFS_PRIVATE_KEY: ${CFS_PRIVATE_KEY}
```
Comment (praju-aot) [inline docker-compose.yml:203]: Please update the scheduler readme file with the latest env variables

File: scheduler/src/modules/cgi-sftp/cgi-sftp.controller.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      type: 'object',
+      properties: {
+        file: {
+          type: 'string',
+          format: 'binary',
+        },
+      },
+    },
+  })
+  upload(
```
Comment (krishnan-aot) [inline scheduler/src/modules/cgi-sftp/cgi-sftp.controller.ts]: Is this controller needed in the real app? I thought this is just a scheduled job.

File: scheduler/src/modules/cgi-sftp/cgi-sftp.service.ts (hunk 1 above)
Comment (praju-aot) [reply scheduler/src/modules/cgi-sftp/cgi-sftp.service.ts:14]: If logger isn't configured in scheduler component, I would advise to add logging as a new ticket

Comment (gchauhan-aot) [reply scheduler/src/modules/cgi-sftp/cgi-sftp.service.ts:14]: yes, this is just to run the set up locally, all of us will have to add public private key locally to mock sftp server and to run the container locally. not needed in OpenShift environments.

File: docker-compose.yml (hunk 2 above)
Comment (gchauhan-aot) [reply docker-compose.yml]: yes, this is just to run the set up locally, all of us will have to add public private key locally to mock sftp server and to run the container locally. not needed in OpenShift environments. I will be posting all the steps to mock sftp container in dev channel

File: scheduler/src/modules/cgi-sftp/cgi-sftp.controller.ts (hunk 4 above)
Comment (gchauhan-aot) [reply scheduler/src/modules/cgi-sftp/cgi-sftp.controller.ts]: Added a note to PR as well. this is just for testing right now, as create file functionality is not ready. Will remove the controller as soon as create file functionality is ready and scheduled.

File: scheduler/src/modules/cgi-sftp/cgi-sftp.service.ts (hunk 1 above)
Comment (gchauhan-aot) [reply scheduler/src/modules/cgi-sftp/cgi-sftp.service.ts]: sure. on it.

Comment (gchauhan-aot) [reply scheduler/src/modules/cgi-sftp/cgi-sftp.service.ts]: done

File: docker-compose.yml (hunk 3 above)
Comment (gchauhan-aot) [reply docker-compose.yml:203]: updated

**Review by krishnan-aot — APPROVED** [review]

**Pull request merged**